model/project_parser.go

```diff
@@ -465,12 +468,15 @@ func LoadProjectForVersion(v *Version, identifier string, shouldSave bool) (*Pro
 
 	// TODO: don't need separate ppFromDB variable once UseParserProject = true
 	if shouldSave && ppFromDB == nil {
-		if err = pp.Insert(); err != nil {
+		// we upsert here instead of insert bc it is possible that since this function started a project has been inserted, and we don't want to error in that case
```

Comment:
We want to return `mongo.ErrNoDocuments` in that case, right? (which I assume we do, but the comment could be slightly clearer, because generate.tasks is confusing for someone reading it for the first time)

Reply:
I kind of think we don't want to return an error: the purpose of this insert here is just to insert the parser project because the document shouldn't exist yet, but in the case of a race the document may have been inserted while the yaml was being parsed or something. 

I switched to upsert because it will already return nil if no documents are found, but maybe this isn't very clear: we could try to insert and just suppress the error if it's a duplicate key error? Or before inserting check again that the document doesn't exist already (I prefer the former personally).

Reply:
Ah, I see. I missed that it was `UpsertWithConfigNumber`. I think that makes sense here.

Reply:
I'm actually going to change it for clarity and to not over-complicate UpsertWithConfigNumber because that's now different.